rand32_ctr_drbg.c fails to build

When I try:

`./scripts/sanitycheck -p frdm_kw41z -s samples/net/sockets/websocket_client/sample.net.sockets.websocket_client`

I get:

```
/home/galak/git/zephyr/subsys/random/rand32_ctr_drbg.c: In function 'ctr_drbg_initialize':
/home/galak/git/zephyr/subsys/random/rand32_ctr_drbg.c:72:9: error: passing argument 3 of 'mbedtls_ctr_drbg_seed' discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
   72 |         entropy_driver,
      |         ^~~~~~~~~~~~~~
In file included from /home/galak/git/zephyr/subsys/random/rand32_ctr_drbg.c:19:
/home/galak/git/modules/crypto/mbedtls/include/mbedtls/ctr_drbg.h:278:26: note: expected 'void *' but argument is of type 'const struct device *'
  278 |                    void *p_entropy,
      |                    ~~~~~~^~~~~~~~~

```

--- a/subsys/random/rand32_ctr_drbg.c
+++ b/subsys/random/rand32_ctr_drbg.c
@@ -37,7 +37,7 @@ static mbedtls_ctr_drbg_context ctr_ctx;
 
 static int ctr_drbg_entropy_func(void *ctx, unsigned char *buf, size_t len)
 {
-	return entropy_get_entropy(ctx, (void *)buf, len);
+	return entropy_get_entropy(entropy_driver, (void *)buf, len);
 }
 
 #elif defined(CONFIG_TINYCRYPT)
@@ -69,7 +69,7 @@ static int ctr_drbg_initialize(void)
 
 	ret = mbedtls_ctr_drbg_seed(&ctr_ctx,
 				    ctr_drbg_entropy_func,
-				    entropy_driver,
+				    NULL,
 				    drbg_seed,
 				    sizeof(drbg_seed));
 

